### PR TITLE
[IOTDB-875]fix flush storage group error when no data in storage group

### PR DESCRIPTION
**Pull request by neuyilan**

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/engine/StorageEngine.java [hunk 1]
```diff
@@ -394,6 +413,15 @@ public void asyncCloseProcessor(PartialPath storageGroupPath, boolean isSeq)
         processor.writeUnlock();
       }
     } else {
+      // bug fix, for storage group do does not have data
+      // https://issues.apache.org/jira/browse/IOTDB-875
+      List<PartialPath> allStorageGroupPaths = MManager.getInstance()
```
Comment (Genius-pig) [inline server/src/main/java/org/apache/iotdb/db/engine/StorageEngine.java:418]: can you use `getStorageGroupPath` instead of getting all storage groups?

Comment (neuyilan) [reply server/src/main/java/org/apache/iotdb/db/engine/StorageEngine.java:418]: sure, thanks for your advice.

File: server/src/test/java/org/apache/iotdb/db/integration/IoTDBFlushQueryMergeIT.java [hunk 2]
```diff
@@ -154,4 +154,24 @@ public void testFlushGivenGroup() throws ClassNotFoundException {
...
+        .getConnection(Config.IOTDB_URL_PREFIX + "127.0.0.1:6667/", "root", "root");
+        Statement statement = connection.createStatement()) {
+      statement.execute("SET STORAGE GROUP TO root.nodatagroup1");
+      statement.execute("SET STORAGE GROUP TO root.nodatagroup2");
+      statement.execute("SET STORAGE GROUP TO root.nodatagroup3");
+      statement.execute("FLUSH root.nodatagroup1");
+      statement.execute("FLUSH root.nodatagroup2");
+      statement.execute("FLUSH root.nodatagroup3");
+    } catch (Exception e) {
+      e.printStackTrace();
```
Comment (yuqi1129) [inline server/src/test/java/org/apache/iotdb/db/integration/IoTDBFlushQueryMergeIT.java:172]: `e.printStackTrace()` can be replaced with logger

Comment (neuyilan) [reply server/src/test/java/org/apache/iotdb/db/integration/IoTDBFlushQueryMergeIT.java:172]: Thanks for your friendly reminder.

**Review by qiaojialin — APPROVED** [review]

**Pull request merged**